File: lab1 stack, queue, deque/E.py
f = [int(i) for i in input().split()]
s = [int(i) for i in input().split()]

i = 0
while f and s:
    if f[0] == 9 and s[0] == 0:
        s.append(f.pop(0))
        s.append(s.pop(0))
        i += 1
    elif f[0] == 0 and s[0] == 9:
        f.append(f.pop(0))
        f.append(s.pop(0))
        i += 1
    elif f[0] > s[0]:
        f.append(f.pop(0))
        f.append(s.pop(0))
        i += 1
    elif f[0] < s[0]:
        s.append(f.pop(0))
        s.append(s.pop(0))
        i += 1
    if i >= 10 ** 6:
        print('botva')
        exit()

if f and not s:
    print('first', i)
else: print('second', i)

File: lab1 stack, queue, deque/E.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    queue<int> f, s;
    int a;
    int cnt = 0;
    int n = 5;


    for (int i = 0; i < n; i++){
        cin >> a;
        f.push(a);
    }
    for (int i = 0; i < n; i++){
        cin >> a;
        s.push(a);
    }

    while (!f.empty() && !s.empty()){
        if (f.front() == 9 && s.front() == 0){
            s.push(f.front());
            s.push(s.front());
        }else if (f.front() == 0 && s.front() == 9){
            f.push(f.front());
            f.push(s.front());
        }else if (f.front() > s.front()){
            f.push(f.front());
            f.push(s.front());
        }else if (f.front() < s.front()){
            s.push(f.front());
            s.push(s.front());
        }

        s.pop();
        f.pop();
        cnt++;

        if (cnt >= 1000000){
            cout << "botva";
            return 0;
        }
    }

    if (!f.empty() && s.empty())
    cout << "first " << cnt;
    else 
    cout << "second " << cnt;
    
}
